webserver/src/main/java/io/kestra/webserver/services/ai/gemini/GeminiAiService.java

```diff
@@ -0,0 +1,362 @@
+            .instruction(pluginRelevanceInstructions)
+            .outputKey("mostRelevantTypes")
+            .afterAgentCallback(callbackContext -> {
+                String mostRelevantTypes = ((String) callbackContext.state().get("mostRelevantTypes")).trim();
+                if (POSSIBLE_ERROR_MESSAGES.contains(mostRelevantTypes)) {
+                    callbackContext.state().put("responseForUser", mostRelevantTypes);
+                    return shortCircuitLlmResponse(callbackContext, mostRelevantTypes);
+                }
+
+                if (mostRelevantTypes.isEmpty()) {
+                    return Maybe.empty();
+                }
+
+                String[] identifiedTypes = mostRelevantTypes.split(" ?, ?");
+
+                JsonNode minifiedSchema = JacksonMapper.ofJson().convertValue(jsonSchemaGenerator.schemas(Flow.class, false, Optional.of(Arrays.asList(identifiedTypes))), JsonNode.class);
+                minifySchema(minifiedSchema);
+                try {
+                    PluginRegistry registry = pluginRegistry;
+                    callbackContext.state().put(
+                        "flowSchema",
+                        JacksonMapper.ofJson().writeValueAsString(minifiedSchema)
+                    );
+                } catch (JsonProcessingException e) {
+                    log.error("Failed to serialize flow schema for AI Agent", e);
+                }
+
+                return Maybe.empty();
+            })
+            .build();
+    }
+
+    private static LlmAgent createYamlBuilderAgent(Gemini model) {
+        return LlmAgent.builder()
+            .name("kestra_init_flow_builder_agent")
+            .description("An agent to bootstrap a Kestra Flow YAML.")
+            .model(model)
+            .generateContentConfig(GenerateContentConfig.builder().temperature(0.9f).maxOutputTokens(50000).build())
+            .outputKey("responseForUser")
+            .beforeAgentCallback((beforeAgentCallback) -> {
+                Object responseForUser = beforeAgentCallback.state().get("responseForUser");
+                if (responseForUser != null) {
+                    return shortCircuitLlmResponse(beforeAgentCallback, responseForUser.toString());
+                }
+
+                return Maybe.empty();
+            })
+            .instruction(
+                """
+                    You are part of a Kestra Flow builder service. You are an expert in generating Kestra Flow YAML. Your task is to generate a valid Kestra Flow YAML based strictly on the following schema:
+                    ```
+                    {{flowSchema}}
+                    ```
+                    
+                    Whenever we're speaking about examples, properties or outputs in the below rules, they are strictly based on the above schema, any external source for those must be discarded because they are not a source of truth.
```

Comment:
```suggestion
                    All examples, properties, or outputs discussed in the guidelines below are strictly based on the above schema, so make sure to strictly follow that schema and use ONLY information provided in the current context, nothing else.
```
Redundant, we've already said: "Your task is to generate a valid Kestra Flow YAML **strictly following the schema below**" so I'd cut this sentence entirely, but if you feel strongly to keep it, just keep it shorter as suggested. 

The "Discard any external source" is confusing as we're not fetching any external sources (I'm aware you wanted to say: discard your existing base knowledge and don't hallucinate) but saying to strictly follow the schema and context provided is enough.

Reply:
I will test without and see if it's enough, I did many experiments so maybe along with all these rephrasing it's not needed anymore.